### ISSUE TITLE
ADS on Mac fails to connect with a new or edited connection

### ISSUE THREAD

**Issue by cheenamalhotra** [microsoft/azuredatastudio#22767]
Content copied from: https://learn.microsoft.com/en-us/answers/questions/1194048/azure-data-studio-on-mac-just-spins-but-works-on-w

All latest versions on Mac fail to connect with a new or edited connection, but the latest Windows version works fine.

The most recent installed on local laptop that fails is: 1.42.0 (Universal)

ADS in Windows connects immediately with the same connection string and settings.

The behavior is it just spins when trying to connect to my database on a new or edited connection. When trying to connect to an existing/cached connection that normally works, it just spins, but closing the app and restarting allows for an existing connection to succeed. Trying to create a new connection, or editing a connection gets me spinning again, until I close the app.

I see no errors, just spinning - viewing all views in the OUTPUT tab.

Question: What can I do to see why it spins?

Version: 1.42.0 (Universal)
Commit: 22f384e569bc90d8b3231cc8c8cff99ea16c5052
Date: 2023-03-17T18:04:34.407Z (1 wk ago)
VS Code: 1.67.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 22.3.0
----------

[20230414T100057.zip](https://github.com/microsoft/azuredatastudio/files/11234911/20230414T100057.zip)
Oury requested my log files from my Mac to troubleshoot my inability to connect to my on-prem DB (works perfectly on Windows version of ADS.
Our thread is here:
https://learn.microsoft.com/en-us/answers/questions/1194048/azure-data-studio-on-mac-just-spins-but-works-on-w

_Originally posted by @tjhee in https://github.com/microsoft/azuredatastudio/issues/21948#issuecomment-1508912339_

**Comment by cheenamalhotra** [comment]
Hi @tjhee 

Opened this issue to understand your case better. I believe the use-case involves connecting to an on-premise SQL server from Mac OS. From the logs you provided (`sqltools.log`), I see below command fails to execute which leads to exception being swallowed and session not completing:

```
Verbose: 0 : eventTID:1026422 SqlCommand.Set_Connection | API | ObjectId 3, Client Connection Id 50652014-e8ed-4994-a830-3f671073bf2b
Verbose: 0 : eventTID:1026422 <prov.DbConnectionHelper.ConnectionString_Get|API> 1
Verbose: 0 : eventTID:1026422 SqlCommand.Set_CommandTimeout | API | ObjectId 3, Command Timeout value 60, Client Connection Id 50652014-e8ed-4994-a830-3f671073bf2b
Verbose: 0 : eventTID:1026422 SqlCommand.Set_CommandText | API | Object Id 3, String Value = 'SELECT SERVERPROPERTY('EngineEdition'), SERVERPROPERTY('productversion'), SERVERPROPERTY ('productlevel'), SERVERPROPERTY ('edition'), SERVERPROPERTY ('MachineName'), SERVERPROPERTY ('ServerName'), (SELECT CASE WHEN EXISTS (SELECT TOP 1 1 from [sys].[all_columns] WITH (NOLOCK) WHERE name = N'xml_index_type' AND OBJECT_ID(N'sys.xml_indexes') = object_id) THEN 1 ELSE 0 END AS SXI_PRESENT)', Client Connection Id 50652014-e8ed-4994-a830-3f671073bf2b
Verbose: 0 : eventTID:1026422 SqlCommand.Set_CommandType | API | ObjectId 3, Command type value 1, Client Connection Id 50652014-e8ed-4994-a830-3f671073bf2b
Verbose: 0 : eventTID:1026422 <sc.SqlConfigurableRetryLogicManager.CommandProvider|INFO> Requested the CommandProvider value.
Verbose: 0 : eventTID:1026422 SqlCommand.RunExecuteReaderTds | Info | Object Id 3, Activity Id 9899e571-cec2-42b0-a887-046186867066:1, Client Connection Id 50652014-e8ed-4994-a830-3f671073bf2b, Command executed as SQLBATCH, Command Text 'SELECT SERVERPROPERTY('EngineEdition'), SERVERPROPERTY('productversion'), SERVERPROPERTY ('productlevel'), SERVERPROPERTY ('edition'), SERVERPROPERTY ('MachineName'), SERVERPROPERTY ('ServerName'), (SELECT CASE WHEN EXISTS (SELECT TOP 1 1 from [sys].[all_columns] WITH (NOLOCK) WHERE name = N'xml_index_type' AND OBJECT_ID(N'sys.xml_indexes') = object_id) THEN 1 ELSE 0 END AS SXI_PRESENT)' 
Verbose: 0 : eventTID:1026422 TdsParserStateObjectManaged.WritePacket | Info | Session Id 50652014-e8ed-4994-a830-3f671073bf2b, SendAsync Result 0
Verbose: 0 : eventTID:1026422 TdsParserStateObjectManaged.ReadSyncOverAsync | Info | State Object Id 1, Session Id 50652014-e8ed-4994-a830-3f671073bf2b
Verbose: 0 : eventTID:1026422 SqlError.ctor | ERR | Info Number 0, Error State 0, Error Class 20, Error Message 'A transport-level error has occurred when receiving results from the server. (provider: TCP Provider, error: 0 - Undefined error: 0)', Procedure '', Line Number 0
Verbose: 0 : eventTID:1026422 SqlConnection.OnError | Info | Object Id 1, Connection broken.
```

The below command fails to execute:

`SELECT SERVERPROPERTY('EngineEdition'), SERVERPROPERTY('productversion'), SERVERPROPERTY ('productlevel'), SERVERPROPERTY ('edition'), SERVERPROPERTY ('MachineName'), SERVERPROPERTY ('ServerName'), (SELECT CASE WHEN EXISTS (SELECT TOP 1 1 from [sys].[all_columns] WITH (NOLOCK) WHERE name = N'xml_index_type' AND OBJECT_ID(N'sys.xml_indexes') = object_id) THEN 1 ELSE 0 END AS SXI_PRESENT)`

Can you provide output of this command by running the above command on your server instance?

**Comment by tjhee** [comment]
Yes! Our on-prem SQL Server gives me the following output when running the command you provided:

3	10.50.1600.1	RTM	Enterprise Edition (64-bit)	DATABASE2	DATABASE2\ROPESQLSERVER	0

**Comment by cheenamalhotra** [comment]
**10.50.1600.1** product version looks like **SQL Server 2008 R2**.

ADS probably doesn't officially support this version I think, but I will look into why the command fails, and if it does it should be handled well instead of continuous spinning.

**Comment by tjhee** [comment]
It's too bad that the WindowsOS ADS works fine, but MacOS ADS does not connect to our version. It's understandable, as our on-prem version may be out of support.
*Note: Using PowerShell on MacOS in a terminal window connects without issue to our old, on-prem version. I very much like ADS's convenient UI, however.
Thank you for your support and valuable feedback to my issue.

**Comment by Kenerator** [comment]
> **10.50.1600.1** product version looks like **SQL Server 2008 R2**.
> 
> ADS probably doesn't officially support this version I think, but I will look into why the command fails, and if it does it should be handled well instead of continuous spinning.

I'm currently having this **exact same issue**.

**Output from the productversion query** executed against the same SQL Server from my Windows laptop:
(No column name)	(No column name)	(No column name)	(No column name)	(No column name)	(No column name)	(No column name)
3	16.0.4165.4	RTM	Developer Edition (64-bit)	dg-db-01	dg-db-01	1

**ADS Version info:Version:** 
1.50.0
Commit: 49c73a84388731fec941dac8d938345ac16bd470
Date: 2024-11-17T22:25:36.863Z
VS Code: 1.82.0
Electron: 30.5.1
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Darwin arm64 23.6.0

No problem connecting from my Windows laptop.
No problem connecting from my MacBook Pro using DBeaver.

I hope that narrows things down some for all y'all. I would like to use ADS on both laptops, but at least since DBeaver is working for me on both Windows and MacOS, it's not a complete blocker.
